cmd/fyne_demo/screens/widget.go

```diff
@@ -289,6 +289,37 @@ func stopProgress() {
 	endProgress <- struct{}{}
 }
 
+func makeListTab() fyne.CanvasObject {
+	var data []string
+	for i := 0; i < 1000; i++ {
+		data = append(data, fmt.Sprintf("Test Item %d", i))
+	}
+
+	icon := widget.NewIcon(nil)
+	label := widget.NewLabel("Select An Item From The List")
+	hbox := fyne.NewContainerWithLayout(layout.NewHBoxLayout(), icon, label)
+
+	list := widget.NewList(
+		func() int {
+			return len(data)
+		},
+		func() fyne.CanvasObject {
+			return fyne.NewContainerWithLayout(layout.NewHBoxLayout(), widget.NewIcon(theme.DocumentIcon()), widget.NewLabel("Template Object"))
+		},
+		func(index int, item fyne.CanvasObject) {
+			item.(*fyne.Container).Objects[1].(*widget.Label).SetText(data[index])
+		},
+	)
+	list.OnItemSelected = func(index int, item fyne.CanvasObject) {
+		icon.Resource = item.(*fyne.Container).Objects[0].(*widget.Icon).Resource
+		label.Text = data[index]
+		icon.Refresh()
+		label.Refresh()
```

Comment:
Consider using the Setters which call Refresh for you

Reply:
This should be resolved now